### PR TITLE
Add Open Liberty to devfile doc

### PR DESCRIPTION
**Pull request by awisniew90**
Signed-off-by: Adam Wisniewski <awisniew@us.ibm.com>

**What type of PR is this?**

/kind documentation
[skip ci]

**What does does this PR do / why we need it**:

Add's an Open Liberty example to "deploying-a-devfile-using-odo.adoc"

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [x] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

### CODE REVIEW

**Comment by girishramnani** [comment]
is this still WIP @awisniew90 ?

File: docs/public/deploying-a-devfile-using-odo.adoc [hunk 1]
```diff
@@ -398,3 +398,102 @@ Run `odo delete` to delete the application from cluster.
...
+[source,sh]
+----
+ $ git clone https://github.com/OpenLiberty/application-stack-intro.git && cd application-stack-intro
+----
+
+. Create an Open Liberty odo component
++
+[source,sh]
+----
+   $ odo create java-openliberty myopenliberty
```
[the remark this answers is not in the document]
Comment (awisniew90) [reply docs/public/deploying-a-devfile-using-odo.adoc]: ✅

File: docs/public/deploying-a-devfile-using-odo.adoc [hunk 2]
```diff
@@ -398,3 +398,102 @@ Run `odo delete` to delete the application from cluster.
...
+   
+   Validation
+    ✓  Checking devfile existence [29496ns]
+    ✓  Checking devfile compatibility [60602ns]
+    ✓  Creating a devfile component from registry: DefaultDevfileRegistry [62676ns]
+    ✓  Validating devfile component [229548ns]
+
+   Please use `odo push` command to create the component with source deployed
```
[the remark this answers is not in the document]
Comment (awisniew90) [reply docs/public/deploying-a-devfile-using-odo.adoc:423]: ✅

**Review by scottkurz — APPROVED** [review]
Looks good.

**Comment by awisniew90** [comment]
@dharmit - good point. Made the change in this PR.

File: docs/public/deploying-a-devfile-using-odo.adoc [hunk 3]
```diff
@@ -398,3 +398,102 @@ Run `odo delete` to delete the application from cluster.
...
+. Have odo watch for changes in the source code
++
+[source,sh]
+----
+ $ odo watch
+----
+
+You can now continue developing your application. Just refreshing the browser will render the source code changes.
+
+. To delete your deployed application:
```
Comment (mik-dass) [inline docs/public/deploying-a-devfile-using-odo.adoc:485]: We should add `Run odo delete to delete the application from cluster.` before this list just like in the previous example.

**Review by mik-dass — APPROVED** [review]
/approve

**Comment by dharmit** [comment]
CI failure is attributed to #3256 
```
 [odo] I1202 17:54:07.764554   68819 deployments.go:139] Waiting for deployment "rljykk" rollout to finish: 0 of 1 updated replicas are available...
[odo] I1202 17:54:07.764561   68819 deployments.go:146] Waiting for deployment spec update to be observed...
[odo]  ✗  Waiting for component to start [5m]
[odo]  ✗  Failed to start component with name rljykk. Error: Failed to create the component: error while waiting for deployment rollout: timeout while waiting for rljykk deployment roll out
Deleting project: ytiuvnsisr
Running odo with args [odo project delete ytiuvnsisr -f]
[odo] I1202 17:59:07.097164   68844 util.go:736] HTTPGetRequest: https://raw.githubusercontent.com/openshift/odo/master/build/VERSION
[odo] I1202 17:59:07.097468   68844 util.go:757] Response will be cached in /tmp/odohttpcache for 1h0m0s
[odo]  ⚠  Warning! Projects are deleted from the cluster asynchronously. Odo does its best to delete the project. Due to multi-tenant clusters, the project may still exist on a different node.
[odo] I1202 17:59:07.280852   68844 odo.go:72] Could not get the latest release information in time. Never mind, exiting gracefully :)
[odo]  ✓  Deleted project : ytiuvnsisr
Setting current dir to: /go/src/github.com/openshift/odo/tests/integration/operatorhub
Deleting dir: /tmp/112090629
------------------------------
• Failure [363.874 seconds]
odo service command tests for OperatorHub
/go/src/github.com/openshift/odo/tests/integration/operatorhub/cmd_service_test.go:17
  When linking devfile component with Operator backed service
  /go/src/github.com/openshift/odo/tests/integration/operatorhub/cmd_service_test.go:360
    should successfully connect and disconnect a component with an existing service [It]
[Fail] odo service command tests for OperatorHub When linking devfile component with Operator backed service [It] should successfully connect and disconnect a component with an existing service 
/go/src/github.com/openshift/odo/tests/helper/helper_run.go:34
Ran 15 of 15 Specs in 736.820 seconds
FAIL! -- 14 Passed | 1 Failed | 0 Pending | 0 Skipped 
```

**Pull request merged**